clients/githubrepo/graphql.go

```diff
@@ -191,15 +198,26 @@ func (handler *graphqlHandler) isArchived() (bool, error) {
 	return handler.archived, nil
 }
 
-// nolint: unparam
 func commitsFrom(data *graphqlData, repoOwner, repoName string) ([]clients.Commit, error) {
 	ret := make([]clients.Commit, 0)
 	for _, commit := range data.Repository.Object.Commit.History.Nodes {
 		var committer string
-		if commit.Committer.User.Login != nil {
+		// Find the commit's committer.
+		if commit.Committer.User.Login != nil && *commit.Committer.User.Login != "" {
 			committer = *commit.Committer.User.Login
+		} else if commit.Committer.Name != nil &&
+			// Username "GitHub" may indicate the commit was committed by GitHub.
+			// We verify that the commit is signed by GitHub, because the name can be spoofed.
+			*commit.Committer.Name == "GitHub" &&
+			commit.Signature.IsValid &&
+			commit.Signature.WasSignedByGitHub {
+			committer = "github"
 		}
-		// TODO(#1543): Figure out a way to safely get committer if `User.Login` is `nil`.
+
+		if committer == "" {
+			return ret, fmt.Errorf("commit %s: %w", commit.Oid, errorInvalidCommitterLogin)
```

Comment:
Why not leave `committer` as `nil` if this is the case? This might cause a lot of errors in the cron job.

Reply:
we don't need to expose `wasSignedByGitHub`, so that checks keep using the committer's login.
Let's try to see what happens in the integration tests before using this code for the weekly cron. It'd be great to be able to report the committer's login reliably.

Reply:
so I'll keep code as-is. Hope this does not backfire

Reply:
I'm running with the latest main code, and I think this is causing an error for me:
```shell
$ ./scorecard --repo=dart-lang/sdk --checks=Fuzzing
Starting [Fuzzing]
2022/03/03 19:26:00 internal error: ListCommits:error during graphqlHandler.setup: commit d3e36fab5eb55706a36144fe41abc2546ffa0b55: cannot retrieve committer login
panic: internal error: ListCommits:error during graphqlHandler.setup: commit d3e36fab5eb55706a36144fe41abc2546ffa0b55: cannot retrieve committer login

goroutine 1 [running]:
log.Panic({0xc000667ab8, 0xc0002be1c0, 0x100df38})
        log/log.go:354 +0x65
github.com/ossf/scorecard/v4/cmd.rootCmd(0xc000470000)
        github.com/ossf/scorecard/v4/cmd/root.go:138 +0x751
github.com/ossf/scorecard/v4/cmd.New.func2(0xc000421680, {0xe9e91a, 0x2, 0x2})
        github.com/ossf/scorecard/v4/cmd/root.go:62 +0x1d
github.com/spf13/cobra.(*Command).execute(0xc000421680, {0xc0000a8160, 0x2, 0x2})
        github.com/spf13/cobra@v1.3.0/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0xc000421680)
        github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.3.0/command.go:902
main.main()
        github.com/ossf/scorecard/v4/main.go:27 +0x25
```

Reply:
Created #1710